### ISSUE TITLE
Train the pretrained embedding for fbwq_full is time consuming using libKGE

### ISSUE THREAD

**Issue by mangonihao** [malllabiisc/EmbedKGQA#97]
When I use the LibKGE to train the RotatE or ComplEx embeddings for fbwq dataset, it runs very slow on my server. So I wonder how much time and what devices you trained these embedding?

**Comment by apoorvumang** [comment]
what is your setup? are you using a gpu?

**Comment by mangonihao** [comment]
![image](https://user-images.githubusercontent.com/35483387/136650521-6d5d89cb-4e0c-4631-88a4-89191f41abe3.png)
![image](https://user-images.githubusercontent.com/35483387/136650539-e967f521-2792-44fd-bfef-7e2257ce3fce.png)
Yes. My device is P40 , which has 22G memory

**Comment by mangonihao** [comment]
> and what config did you use? ie batch size etc.
batch size is 100, lr is 0.2
![image](https://user-images.githubusercontent.com/35483387/136650821-503f42ce-752b-4ddb-9b0b-c4193a933376.png)

**Comment by mangonihao** [comment]
this is my setting:
`job.type: train
dataset.name: fbwq_full
train:
  max_epochs: 1000
  optimizer.default:
    type: Adagrad
    args:
      lr: 0.2
  lr_scheduler: ReduceLROnPlateau
  lr_scheduler_args:
    mode: max
    patience: 4

model: complex
lookup_embedder:
  dim: 100
  regularize_weight: 0.8e-7
  initialize: normal_
  initialize_args:
    normal_:
      mean: 0.0
      std: 0.1
`
I don't set this. Maybe it use KvsALL by default.
this is your setting:
import: [lookup_embedder]

complex:
  class_name: ComplEx
  entity_embedder:
    type: lookup_embedder
    +++: +++
  relation_embedder:
    type: lookup_embedder
    +++: +++
I can't find where to set this.

**Comment by apoorvumang** [comment]
Take a look at this config, its for WikiData5M from the official libkge github repo

http://web.informatik.uni-mannheim.de/pi1/libkge-models/wikidata5m-complex.yaml

It uses negative sampling. You should be able to modify it for your needs. If it works for you, I would request you share it here as well so I can add it to our repo

**Comment by mangonihao** [comment]
The libKGE doesnot implement Tucker, so I used the official repo to train Tucker on your fbwq_full dataset. But it's really slow, it takes 6 hours to finish one epoch. I wonder whether you tried Tucker on fbwq_full dataset ?
